src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/RSAPkcs1X509SignatureGenerator.cs

```diff
@@ -32,9 +33,16 @@ internal static PublicKey BuildPublicKey(RSA rsa)
         {
             RSAParameters parameters = rsa.ExportParameters(false);
 
-            byte[] rsaPublicKey = DerEncoder.ConstructSequence(
-                DerEncoder.SegmentedEncodeUnsignedInteger(parameters.Modulus),
-                DerEncoder.SegmentedEncodeUnsignedInteger(parameters.Exponent));
+            byte[] rsaPublicKey;
+
+            using (AsnWriter writer = new AsnWriter(AsnEncodingRules.DER))
+            {
+                writer.PushSequence();
+                writer.WriteIntegerUnsigned(parameters.Modulus);
+                writer.WriteIntegerUnsigned(parameters.Exponent);
+                writer.PopSequence();
+                rsaPublicKey = writer.Encode();
+            }
```

Comment:
This doesn't even need the AsnWriter now.  `rsaPublicKey = rsa.ExportRSAPublicKey();`